raster/src/test/scala/geotrellis/raster/geotiff/GeoTiffRasterSourceSpec.scala

```diff
@@ -85,6 +86,14 @@ class GeoTiffRasterSourceSpec extends AnyFunSpec with RasterMatchers with GivenW
     }
   }
 
+  it("should preserve baseTiff on resample/reproject") {
+    val resampled = source.resample((source.cols * 0.95).toInt , (source.rows * 0.95).toInt, NearestNeighbor)
+    val reprojected = resampled.reproject(WebMercator)
+    assert(source.tiff == reprojected.asInstanceOf[GeoTiffReprojectRasterSource].tiff)
```

Comment:
Could we use `assertEqual` instead? The regular assert may also work, but let's try to preserve the Spec style!

Sadly it's a bit verbose; smth like https://github.com/locationtech/geotrellis/blob/6aa9383dd4795575bff9f8bb8e2d29449b7c1599/raster/src/test/scala/geotrellis/raster/io/geotiff/writer/GeoTiffWriterSpec.scala#L282-L290

Reply:
or, right, we need that by reference comparison, but let's still use the scala test syntax:

```scala
source.tiff shouldBe reprojected.asInstanceOf[GeoTiffReprojectRasterSource].tiff
val converted = reprojected.convert(UShortCellType)
source.tiff shouldBe converted.asInstanceOf[GeoTiffReprojectRasterSource].tiff
```